Feature/removed the usage of block size param

YarikRevich opened the pull request:
## Fixes

Removes usage of "block_size" param

Referes to issue #1505

YarikRevich committed:
feature: removed the usage of block size param

boddumanohar commented on code/go/0chain.net/sharder/chain.go:
```diff
@@ -35,10 +34,7 @@ func SetupSharderChain(c *chain.Chain) {
-	if transactionCacheSize > 5000 {
-		transactionCacheSize = 5000
-	}
+	transactionCacheSize := 5 * blockCacheSize
```
irrespective of the blocksize, we are setting the transaction cache size as 5 times the block cache size (which is 500). Why 5 times?

YarikRevich replied:
I discussed this with @dabasov. He said, that in general transactionCacheSize is not mandatory to be calculated in the way it did before. So there may be any constant

YarikRevich committed:
Merge branch 'staging' of https://github.com/0chain/0chain into feature/remove_max_block_size

YarikRevich committed:
fix: removed dead code

dabasov commented on code/go/0chain.net/miner/miner/worker.go:
```diff
@@ -54,35 +54,12 @@ func TransactionGenerator(c *chain.Chain, workdir string) {
 	}
 
-	switch {
-	case blockSize <= 10:
```
looks like we should calculate `blockSize` as `totalCost/txnCost`
`totalCost` is in config  `    max_block_cost: 10000 #equal to 100ms`
for txnCost it is a little bit more trickier, but looks like we only generate transfer transactions and now we have cost for it in configs `    transfer_cost: 10`

dabasov commented on code/go/0chain.net/miner/protocol_block.go:
```diff
@@ -787,12 +787,12 @@ func (tii *TxnIterInfo) checkForCurrent(txn *transaction.Transaction) {
+func newTxnIterInfo() *TxnIterInfo {
 	return &TxnIterInfo{
 		clients:    make(map[string]*client.Client),
-		eTxns:      make([]datastore.Entity, 0, blockSize),
```
we can estimate blockSize maximum value the same way

YarikRevich committed:
fix: fixed block size generation

peterlimg commented on code/go/0chain.net/miner/miner/worker.go:
```diff
@@ -105,7 +95,7 @@ func TransactionGenerator(c *chain.Chain, workdir string) {
 	trng := rand.New(ts)
 	for {
-		numTxns = trng.Int31n(blockSize)
+		numTxns = trng.Int31n(100)
```
Use the blockSize calculated above

peterlimg commented on code/go/0chain.net/miner/protocol_block.go:
```diff
@@ -592,7 +592,9 @@ func (mc *Chain) updateFinalizedBlock(ctx context.Context, b *block.Block) {
 	}
 
-	tii := newTxnIterInfo(mc.BlockSize())
+	blockSize := mc.ChainConfig.MaxBlockCost() / mc.ChainConfig.TxnTransferCost()
```
blockSize here is the number of txns, so it is `len(b.Txns)`.

peterlimg commented on code/go/0chain.net/miner/protocol_block.go:
```diff
@@ -873,10 +873,12 @@ func (mc *Chain) generateBlock(ctx context.Context, b *block.Block,
-	b.Txns = make([]*transaction.Transaction, 0, mc.BlockSize())
+	b.Txns = make([]*transaction.Transaction, 0, 100)
+
+	initialBlockSize := mc.ChainConfig.MaxBlockCost() / mc.ChainConfig.TxnTransferCost()
```
The cost of `transfer` transaction is not the average cost of all kinds of transactions. So let's not use it to calculate the block size. The initialBlockSize could be the same as `b.Txns` , it will be extended when the capacity is reached.

YarikRevich committed:
fix: fixed incorrect usage of blockSize param

dabasov committed:
Merge remote-tracking branch 'github/staging' into staging

dabasov committed:
Merge branch 'staging' into feature/remove_max_block_size

dabasov reviewed (approved):
lgtm

peterlimg reviewed (approved):
lgtm, please fix the unit tests

YarikRevich committed:
fix: fixed miner unit tests

YarikRevich committed:
Merge branch 'feature/remove_max_block_size' of https://github.com/0chain/0chain into feature/remove_max_block_size

peterlimg committed:
Merge branch 'staging' into feature/remove_max_block_size

peterlimg committed:
Format and cap fix

service-0chain commented:
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | feature/remove_max_block_size  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

peterlimg committed:
Merge branch 'staging' into feature/remove_max_block_size

* staging:
  removes unneeded m1 flag from readme
  remove benchmark notification from staging (#1747)
  Fix empty index when updating challenge pool
  Add pagination limit errors
  Passin only the challenge completion time instead of conf object
  suggested changes
  simplifyig syntax
  reverting config chages
  getting config from MPT
  getting config from MPT
  Dropping all token in write pool
  Emit events on collect-reward for validators

peterlimg committed:
Fix redis/miner panic

Transaction generator is for testing purpose, we should not generate testing transactions when txn_generation.max_transactions is 0
